File: src/main/frontend/src/app/services/review.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { TokenStorageService } from './token-storage.service';

@Injectable({
  providedIn: 'root'
})
export class ReviewService {

  private reviewsUrl: string;
  
  constructor(private http: HttpClient,
    private tokenStorageService: TokenStorageService) {
    this.reviewsUrl = "http://localhost:8080/reviews/";
  }

  upvote(reviewId: number) {
    let userId = this.tokenStorageService.getUser().id;
    let requestBody = {
      "voterId": userId,
      "reviewId": reviewId
    }
    return this.http.post<null>(this.reviewsUrl + "upvoteReview", requestBody);
  }

  downvote(reviewId: number) {
    let userId = this.tokenStorageService.getUser().id;
    let requestBody = {
      "voterId": userId,
      "reviewId": reviewId
    }
    return this.http.post<null>(this.reviewsUrl + "downvoteReview", requestBody);
  }

  addComment(reviewId: number, content: string) {
    let requestBody = {
      "reviewId": reviewId,
      "posterId": this.tokenStorageService.getUser().id,
      "content": content,
    }
    return this.http.post<null>(this.reviewsUrl + "addReply", requestBody);
  }

  deleteComment(reviewId: number, commentId: number) {
    let requestBody = {
      "reviewId": reviewId,
      "commentId": commentId
    }
    return this.http.post<null>(this.reviewsUrl + "deleteReply", requestBody);
  }

}

File: src/main/java/com/stage_facile/stage_facile/services/ReviewService.java
package com.stage_facile.stage_facile.services;

import java.util.Optional;

import com.stage_facile.stage_facile.models.Comment;
import com.stage_facile.stage_facile.models.Internship;
import com.stage_facile.stage_facile.models.Review;

public interface ReviewService {
	public void addReview(Internship internship, Review review);
	public void addComment(Review review, Comment comment);
	public void addComment(Long reviewId, Long userId, String content);	
	public void deleteComment(Review review, Comment comment);
	public void deleteComment(Long reviewId, Long commentId);
	public Optional<Review> find(Long id);
	public void upvote(Long reviewId, Long voterId);
	public void downvote(Long reviewId, Long voterId);
	public void upvoteComment(Long commentId, Long voterId);
	public void downvoteComment(Long commentId, Long voterId);
}
